source/read/specify-documents-to-return.txt

```diff
@@ -0,0 +1,200 @@
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      { "_id" : { "$oid" : "..." }, ..., "name" : "(Lewis Drug Store) Locanda Vini E Olii", "restaurant_id" : "40804423" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "101 Restaurant And Bar", "restaurant_id" : "40560108" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "44 Sw Ristorante & Bar", "restaurant_id" : "40698807" }
+      ...
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Zucchero E Pomodori", "restaurant_id" : "41189590" }
+
+.. _cpp-return-documents-skip:
+
+Skip
+----
+
+To skip a specified number of documents before returning your query results, create
+an instance of the ``mongocxx::options::find`` class and set its ``skip`` field.
+Then, pass your ``mongocxx::options::find`` instance as a parameter to the
+``find()`` method.
+
+The following example returns all documents that have a ``borough`` field value
+of ``"Manhattan"`` and skips the first ``10`` documents:
+
+.. io-code-block::
+
+   .. input:: /includes/read/limit-skip-sort.cpp
+      :start-after: start-skip-method
+      :end-before: end-skip-method
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      { "_id" : { "$oid" : "..." }, ...,  "name" : "Cafe Metro", "restaurant_id" : "40363298" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Lexler Deli", "restaurant_id" : "40363426" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Domino'S Pizza", "restaurant_id" : "40363644" }
+      ...
+
+.. _cpp-return-documents-combine:
+
+Combine Limit, Sort, and Skip
+-----------------------------
+
+You can set the ``limit``, ``sort``, and ``skip`` fields of a single
+``mongocxx::options::find`` instance. This allows you to set a maximum
+number of sorted documents to return, skipping a specified number of
+documents before returning.
+
+The following example returns documents that have a ``cuisine`` value of
+``"Italian"``. The results are sorted in ascending ``name`` value order,
+skipping the first ``10`` documents:
+
+.. io-code-block::
+
+   .. input:: /includes/read/limit-skip-sort.cpp
+      :start-after: start-limit-sort-skip
+      :end-before: end-limit-sort-skip
+      :language: cpp
+      :dedent:
+
+   .. output::
+      
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Acqua", "restaurant_id" : "40871070" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Acqua Restaurant", "restaurant_id" : "41591488" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Acqua Santa", "restaurant_id" : "40735858" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Acquista Trattoria", "restaurant_id" : "40813992" }
+      { "_id" : { "$oid" : "..." }, ..., "name" : "Acquolina Catering", "restaurant_id" : "41381423" }
+
+.. note::
+
+   The order in which you call these methods doesn't change the documents
+   that are returned. The driver automatically reorders the calls to perform the
+   sort and skip operations first, and the limit operation afterward.
```

Comment:
Question: Do sort and skip happen at the same time? Which one happens first relative to the other? I think that the order they happen in would change the results of the operation.